connector.rb

```diff
@@ -1,37 +1,2827 @@
+      end,
+
+      help: lambda do |input, picklist_label, connection, webhook_base_url|
+        <<~HTML
+          Creates a job when an event is received from 4me. To set this webhook up,
+          you will need to register the webhook below in 4me under "settings" => "webhooks". <br><br>
+          <b>Webhook endpoint URL</b>
+          <b class="tips__highlight">#{webhook_base_url}</b>
+          More information on how to use 4me automation rules and webhooks can be found on the <a href="https://developer.4me.com//v1/workato_connector/" target="_blank">4me developer pages</a>.
+        HTML
+      end,
 
+      config_fields:
+        [
+          {
+            name: 'event_selection',
+            label: 'Event',
+            control_type: 'select',
+            pick_list: 'webhook_events',
+            optional: false,
+            hint: 'Select a specific event.'
+          },
+          {
+            name: 'webhook_identifier',
+            label: 'Webhook identifier',
+            optional: false,
+            hint: 'The webhook GraphQL identifier.'
+          },
+          {
+            ngIf: 'input.event_selection == "automation_rule"',
+            name: 'payload_output',
+            extends_schema: true,
+            schema_neutral: true,
+            control_type: 'schema-designer',
+            label: 'Payload schema',
+            sticky: true,
+            hint: 'The payload schema as defined in the "automation_rule" webhook event.'
+          }
+        ],
+
+      webhook_key: lambda do |connection, input|
+        input['webhook_identifier']
+      end,
+
+      webhook_notification: lambda do |
+        input,
+        payload,
+        extended_input_schema,
+        extended_output_schema,
+        headers,
+        params,
+        connection,
+        webhook_subscribe_output|
+        if payload['event'] == input['event_selection'] || payload['event'] == 'webhook.verify'
+          {
+            webhook_id: payload['webhook_id'],
+            webhook_nodeID: payload['webhook_nodeID'],
+            account_id: payload['account_id'],
+            account: payload['account'],
+            custom_url: payload['custom_url'],
+            name: payload['name'],
+            event: payload['event'],
+            object_id: payload['object_id'],
+            object_nodeID: payload['object_nodeID'],
+            person_id: payload['person_id'],
+            person_nodeID: payload['person_nodeID'],
+            person_name: payload['person_name'],
+            instance_name: payload['instance_name'],
+            data: payload['payload'],
+            payload: payload['payload']
+          }
+        end
+      end,
+
+      dedup: lambda do |record|
+        "#{record['account_id']}_#{record['event']}_#{record['object_id']}_#{Time.now.to_f}"
+      end,
+
+      output_fields: lambda do |object_definitions|
+        object_definitions['webhook_event']
```

Comment:
This will always be a single record. The record contains the generic data, available in all webhooks and the payload property contains the basic data, or custom-defined data if an `automation_rule` event is used.